### ISSUE TITLE
Windows 11 KB5030310 update -> black screen on login

### ISSUE THREAD

**Issue by KRMalachi** [Open-Shell/Open-Shell-Menu#1717]
### Describe the bug

Disclaimer: I've only seen this on domain-joined Windows 11 Pro computers! 

After installing this patch, and then rebooting, the login screen will come up fine, but after logging in, the user is greeted by a black screen with a cursor, and no other Windows GUI comes up. The shell (explorer.exe) doesn't open. You can get to the Task Manager via CTRL+SHIFT+ESC key combination, and from there can run various tasks via the Run New Task button. I know there are plenty of sites with various "fixes" for the black screen with cursor issue (simple search will show lots of results), but the only one I've been able to get to work is to uninstall that patch in order to get the Windows shell to load after a subsequent reboot.

However, what I've successfully done is to uninstall Open-Shell, install the patch manually. After successfully installing the patch and rebooting, the Windows shell will load after login. Then I was able to reinstall Open-Shell and everything works again, even after reboots.

### Area of issue

Windows Explorer

### To reproduce

Allow the patch to install via Windows Update (Settings>Windows Update), then reboot. After reboot, login and voila - black screen. This assumes Open-Shell is already installed, of course.

### Expected behavior

Expect the patch to install, and the Windows shell to load properly after a reboot.

### Open-Shell version

4.4.191

### Windows version

Windows 11 Pro 22H2

**Comment by ge0rdi** [comment]
I have updated my Win11 22H2 VM with `Open-Shell` 4.4.191 installed and didn't encounter any issue.
Everything is running properly.

> You can get to the Task Manager via CTRL+SHIFT+ESC key combination, and from there can run various tasks via the Run New Task button.

Could you check if `Explorer` is running in such state?
And if not then try to start it (just run `explorer.exe`) to see what will happen.

Do you have any records related to Explorer in event log (like crashes or so)?

**Comment by rskorecki** [comment]
I have exact same bug behavior as OP when allowing installation of KB5030310

**Comment by KRMalachi** [comment]
Interesting you couldn't get your VM to repeat the issue! It's happened on 4 of my computers at work and 1 at home so far. All are domain-joined Win 11 Pro as I said, and all had Open-Shell 4.4.191 installed. They all have different hardware, so those were really the only things common to all of the computers that had this issue. A couple of them had ExplorerPatcher installed. These were my work desktop and laptop, and my home desktop. The other affected computers at work did not have ExplorerPatcher, so as I said, the only commonalities to all the affected machines were Win 11 Pro 22H2, joined to a domain, and Open-Shell v 4.4.191 installed. 

Explorer.exe was not running after login and getting the black screen, and wouldn't start if I tried to run it manually either. The only option that got things back was to do an advanced boot, and uninstall the most recent update from that advanced boot menu. Once that particular update was uninstalled, then Windows would start up normally. As I said I already fixed the issue on all the affected machines by uninstalling Open-Shell, running the update, rebooting and logging into Windows, then reinstalling Open-Shell.

Another thing I checked when the computers were still in the state of logging into a black screen was to open registry editor and go to 
HKLM\SOFTWARE\Microsoft\Windows NT\CurrentVersion\WinLogon and look at the 'Shell' entry, which should have a value of 'explorer.exe' and verified that they did indeed have that as a value. Many sites I found when searching this issue showed setting that value as a fix for the issue, because it had somehow gotten changed. None of the fixes most of the pages in those search results showed fixed the issue, other than to do the advanced boot and uninstall the patch. Of course this was only ever going to be a temporary fix, because at some point Windows Update will just offer it up again, so I had to figure out a more permanent solution. Unfortunately, unless another machine at work has the issue I can't roll any of the machines back and try to reproduce the issue again.

I'm looking at the Event Viewer on my work desktop and do see an event ID 10010 containing the following message:
"Application 'C:\Windows\explorer.exe' (pid 16736) cannot be restarted - Application SID does not match Conductor SID.."
Other than that, I don't see anything in the logs. Hope this is helpful.

**Comment by KRMalachi** [comment]
> I have exact same bug behavior as OP when allowing installation of KB5030310

If it's time sensitive that you get it fixed, follow the 7th "fix" on this page:
https://beebom.com/fix-windows-11-black-screen-after-update/
and once Windows is working properly, uninstall Open-Shell, run the update, reboot and then reinstall Open-Shell, as I described. If it's not time sensitive, maybe you can help ge0rdi with the trouble-shooting that I can't because I've already figured out a workaround for myself?

**Comment by ge0rdi** [comment]
>  joined to a domain

Not sure how it can be related, but that's something I cannot test right now.
@rskorecki Is your machine also joined to a domain?

> The other affected computers at work did not have ExplorerPatcher

Could they have similar customization software (StartAllBack, ReBar, ...)? Or there is nothing like that?

> and wouldn't start if I tried to run it manually either

Does it show some message when you try to run Explorer manually?
Or it simply does nothing?

Could you eventually use [Process Monitor](https://learn.microsoft.com/en-us/sysinternals/downloads/procmon) and obtain log from Explorer starting? Maybe it will reveal why Explorer refuses to start ...

**Comment by KRMalachi** [comment]
No, the other machines didn't have anything else like that installed, only mine did. The others only had Open-Shell installed. Yeah doesn't make sense to me either that being joined to a domain would be part of the problem, but I wanted to list all the things the machines have in common.

Unfortunately, I can no longer help with the troubleshooting, because I've already used my workaround to get the patch installed. I would have posted before running my workaround, but there really was no way for me to be sure Open-Shell was even the problem at first. The only reason I tried that is because it was the only common software that was installed, so it was literally just a try... granted a logical try, but a try nonetheless. Once I knew it worked on one machine, I tried it on a second to verify, and then had to fix the rest so people could work, so unless and until it happens on one of the other Win 11 machines we have, then I won't be able to help with further trouble-shooting.

**Comment by paraworlds2** [comment]
This is a really serious issue. It's now happened on about 10 of my computers/laptops (some are joined to a domain and some are not). Just happened tonight on 4 of my non-domain Windows 11 22H2 computer/laptops that just got the KB5030310 patch. All of them had ExplorerPatcher but they all had older versions of Open-Shell (not the 4.4.191). One had version 4.4.180.

I was able to fix the first few by logging in with a different user account and then update ExplorerPatcher (I remember a couple times when I logged in it told me that ExplorerPatcher had crashed too many times). I have gotten a black screen a few times in the past with Open-Shell but have always been able to reboot or just run explorer.exe to fix it but that no longer works for this latest bug which may be caused by ExploerPatcher. 

Anyway, my kids did a system restore but then the issue just happened again. By the time I got onto it, my login was also affected (doesn't matter how much you reboot), so I solved it by Ctrl-Alt-Del then Task Manager then click "run new task", then checkmark "create this task with administrative privileges" (note I tried once without clicking that and it complained), then type the following line and pressing Enter:
"https://github.com/valinet/ExplorerPatcher/releases/latest/download/ep_setup.exe"

It pulled up Edge (my default browser) but then it asked twice if I wanted to keep the file because it wasn't "commonly downloaded" (click Keep and then click "Keep anyway"). It finally let me launch it, but then it came up with a very long-winded error message from ExplorerPatcher

"An error has occurred while servicing this product. This is most likely caused by one or more of the backup files from a previous update still being in use. Unlocking the files should fix this issue." Then it gave a bunch of suggestions. I actually just launched ep_setup.exe 2 or 3 more times with the same error but then suddenly the taskbar came back and everything worked again with the start button on the left so I know ExplorerPatcher was running. 

I immediately right clicked the task bar, properties, Updates, "Update program and restart Explorer", which actually got the same long-winded error. I rebooted and everything continued to work. I tried "Update program and restart Explorer" but it said there were no updates. My guess is that the installer tries to restart explorer and that's what actually jumpstarted Windows. I wonder if I had just killed explorer.exe from the task manager screen and then ran explorer.exe it would have jumpstarted it too. I rebooted another time and everything is now fine.

Wow, I'm scared that 100 angry people at my company are about to call me about this. They are not administrators of their laptops, so they can't follow these instructions.

We're also getting the flashing white screen but the latest ExplorerPatcher will kill itself after 3 flashes, so at least that's not a call to IT anymore. My techs tell me they usually kill or uninstall ExplorerPatcher to solve the white flashing screen and then update ExplorerPatcher. 

So I think these wonderful products might be coming to an end for me (I've used Open-Shell/ClassicShell for 7 years on a thousand computers with minimal issues now but ExplorerPatcher for less than a year on 100 laptops with lots of issues). Just too much risk unless the developers can combat the Windows upgrades more gracefully. Both products need to have auto-udpaters that just do it in the background without the user having admin rights like other products like Edge/Chrome, etc. do it. Maybe a special exe we can run from task manager as a non-admin to fix itself. Hope this helps someone.

**Comment by paraworlds2** [comment]
I installed ClassicShell (the original product) on about 5,000 computers at my last place (8 years ago) with virtually no issues, but Windows 10 and especially 11 have been much more problematic.

**Comment by sean-mcl** [comment]
I had the same issue on my ASUS machine after installing the KB5030310 update. Had to run a system recovery.
**I don't have open shell installed.**

Edition	Windows 11 Pro
Version	22H2

**Comment by KRMalachi** [comment]
> I had the same issue on my ASUS machine after installing the KB5030310 update. Had to run a system recovery. **I don't have open shell installed.**
> 
> Edition Windows 11 Pro Version 22H2

Well that changes things, doesn't it? Maybe it was a fluke that uninstalling Open-Shell fixed it for me? But why would that work on all my affected computers? Strange stuff...

**Comment by DerekZiemba** [comment]
I'm on Win11 22H2 Education Edition. The PC is not on a domain & my account is a local accounts  
![image](https://github.com/Open-Shell/Open-Shell-Menu/assets/7283693/56afb249-90db-42a0-b71c-f8924db6cd83)

I tried starting the process several times with task manager but was not successful. Also tried getting the settings app `immersivecontrolpanel` to launch to see if I could roll back, but couldn't launch that either.
![image](https://github.com/Open-Shell/Open-Shell-Menu/assets/7283693/4d485ecc-56bb-4132-810f-432f3f3c691c)


What fixed it for me was upon reboot, ExplorerPatcher detected 3 crashes & reset Explorer.  Then I rebooted again & ExplorerPatcher was able to reinitialize it's settings back to the way I had it, but now OpenShell won't hook on system start.  I have to manually launch `Open-Shell Menu Settings` after login to get the start menu back. 


I already have had a unique issue for months now, where Open-Shell won't hook into  Win-Key down to popup the better start menu, requiring me to annoyingly actually click the button until I restart explorer.  I solved this by removing the Open-Shell startup entry and forcing it to start later in logon by placing a shortcut with target to `C:\Program Files\Open-Shell\StartMenu.exe` in `C:\Users\Derek\AppData\Roaming\Microsoft\Windows\Start` 

But now that fix doesn't seem to do anything.  Open-Shell won't start until I manually start it.

**Comment by DerekZiemba** [comment]
> From the logs screenshot it looks like Explorer was crashing in C:\Windows\dxgi.dll module. That's pretty weird because such file should be located in System32 folder normally. So maybe you have some issue with your Windows installation.

Hmm, that does in fact seem to be ExplorerPatcher
![image](https://github.com/Open-Shell/Open-Shell-Menu/assets/7283693/f4defeeb-27d3-4362-b602-92e7ac63d64d)

> I guess it may be better to discuss this issue on ExplorerPatcher forum. As many indications here seem that this is most likely the culprit (ExplorerPatcher is known for crashing in new/updated Windows unless some support is added).

So I can't recall why now, but I believe there was an error at one point that seemed to point to OpenShell which lead me here before ExplorerPatcher.  But now I'm wondering if I just imagined that... 

> Do you have Start automatically for this user setting enabled?

No I unchecked that a while ago so I could do launch it later in the logon process, since it wasn't hooking explorer properly.  My suspicion was that it was/has something to do with PowerToys & it's Keyboard Manager. 

> Maybe try to reinstall Open-Shell.
> Maybe it could be some interference with ExplorerPatcher or other customization software you are using?
> Try to disable/uninstall those to see if it will help.

For the hooking problem + a more severe problem that fixed itself-*ish*, I've done that + reinstalled ExplorerPatcher, DisplayFusion, PowerToys, Dropbox, NVidia drivers, 7zip, ICaros, ie: anything that was dynamically loading into the explorer process. But had no luck. 

The severe problem was that Explorer would crash when clicking the file explorer button on the taskbar.  I was at the point of nearly reinstalling Windows.  But I _really_ didn't want to because it was only a 2 week fresh install (Aug 8, 2023) & I finally nearly had everything setup, after _days_ spent getting it to a level I could comfortably replace my old machine. [My work around for a while there was opening explorer from Open-Shell using the pinned folders shown here](https://github.com/Open-Shell/Open-Shell-Menu/assets/7283693/9ddf1b67-7e66-4f1a-9be3-11c132cfd483).  What eventually got Explorer working was the Folder Options: 
 - unchecking Show recent/frequent used files/folders & Show files from Office.com
 - clearing the file explorer history
 - checking "Launch folder windows in a separate process". 
   - this was more of an intermediate fix so all of explorer wouldn't crash when I opened a folder.  Then I've just never unchecked it since. 

~~One of the remnants to that issue that still lingers is I can't do Multiple Desktops or get the Task view to work.~~ NVM!! That preview update may have fixed that! It's been broken pretty much since Aug & I haven't gave it a try recently.  Still don't have Widgets though & OneDrive is still borked.  
Sometimes OneDrive will make a brief appearance & massively lag FileExplorer until I click the icon, then it disappears like it was never there. I think it's a conflict between Enterprise & Personal + a system restore I was forced to do after signing into Enterprise. Doing so unexpectedly caused Corporate policies to apply to my machine.  Since reverting all that OneDrive's been borked & this was also the what lead to much of the broken Explorer functionality.  


> What version of Open-Shell do you have?

* Open-Shell 4.4.191
* Explorer-Patcher: 22621.2361.58.1 (literally just updated a few hours ago after the problem)
* DisplayFusion 10.1.2 (Beta 3)

At this point it works well enough that I'm just gonna hope it all works itself out.  I've had good luck with that in the past.

**Comment by DerekZiemba** [comment]
After some more time with it, I have other problems that I don't believe are Open-Shells fault.  Though, it all started with KB5030310 & I have observed the issues presented in this thread.  
The issue is entries under `Run` keys such as  `HKCU\Software\Microsoft\Windows\CurrentVersion\Run`, `HKCU\Software\Microsoft\Windows\CurrentVersion\Policies\Explorer\Run`, `HKLM\SOFTWARE\Microsoft\Windows\CurrentVersion\Run`, `HKLM\SOFTWARE\WOW6432Node\Microsoft\Windows\CurrentVersion\Run` no longer execute. This appears to be why Open-Shell is no longer hooking - it was just the most obvious thing that didn't start up after the issue I had earlier. 

As a temporary workaround, I moved everything I want to startup to `%APPDATA%\Microsoft\Windows\Start Menu\Programs\Startup`, but to my disappointment, these seemingly are launched one-by-one & only after like >10 minutes after logon (when idling?).   I'm not sure if this is the expected behavior of apps here, I always thought they launched immediately after everything else.  Can anyone confirm?  I'm wondering if explorer is getting hung up on something. 
Actually... I'm semi-sure, otherwise my moving Open-Shell to that startup folder originally wouldn't have worked out. 


With KB5030310 I lost some but gained some. For instance: I had no idea Windows 11 had a volume overlay. I have never once seen it before.  
![volume overlay](https://github.com/Open-Shell/Open-Shell-Menu/assets/7283693/143d8332-5a4f-464f-b3cd-962f304670e4)

**Comment by ge0rdi** [comment]
> This appears to be why Open-Shell is no longer hooking.

Not really.
`Open-Shell` uses other means to start early when Explorer is starting (`Run` keys are processes some time after logon, like 20-30s after).

You should try to remove customization software that is loading into Explorer one by one (I'd start with ExplorerPatcher) to see if there will be any difference. You can use `Autoruns` (from Sysinternals) to see what is registered to run in Explorer.
